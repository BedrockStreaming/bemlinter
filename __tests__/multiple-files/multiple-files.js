const bemlinter = require('./../../src/bemlinter');

describe('Bemlinter of multiple files', () => {
  it('should lint with crossed error', (done) => {
    bemlinter(`${__dirname}/*.scss`, {prefix: ['c-']})
      .then(logs => {
        expect(logs).toMatchSnapshot();
        done();
      });
  });
  
  it('should lint with crossed error', (done) => {
    bemlinter(`${__dirname}/*.scss`, {prefix: ['', 'c-']})
      .then(logs => {
        expect(logs).toMatchSnapshot();
        done();
      });
  });
});