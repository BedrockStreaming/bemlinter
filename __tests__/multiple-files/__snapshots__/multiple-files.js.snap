exports[`Bemlinter of multiple files should lint with crossed error 1`] = `
Object {
  "component": Object {
    "error": Array [
      Object {
        "filePath": "./__tests__/multiple-files/component.scss",
        "line": 8,
        "message": "\".c-other-component\" is incoherent with the file name.",
      },
    ],
    "info": Array [
      Object {
        "filePath": "./__tests__/multiple-files/component.scss",
        "line": null,
        "message": "Parsed",
      },
    ],
  },
  "other-component": Object {
    "error": Array [
      Object {
        "filePath": "./__tests__/multiple-files/component.scss",
        "line": 8,
        "message": "\".c-other-component\" should not be styled outside of its own stylesheet.",
      },
    ],
    "info": Array [
      Object {
        "filePath": "./__tests__/multiple-files/other-component.scss",
        "line": null,
        "message": "Parsed",
      },
    ],
  },
}
`;

exports[`Bemlinter of multiple files should lint with crossed error 2`] = `
Object {
  "component": Object {
    "error": Array [
      Object {
        "filePath": "./__tests__/multiple-files/component.scss",
        "line": 8,
        "message": "\".c-other-component\" is incoherent with the file name.",
      },
    ],
    "info": Array [
      Object {
        "filePath": "./__tests__/multiple-files/component.scss",
        "line": null,
        "message": "Parsed",
      },
    ],
  },
  "other-component": Object {
    "error": Array [
      Object {
        "filePath": "./__tests__/multiple-files/component.scss",
        "line": 8,
        "message": "\".c-other-component\" should not be styled outside of its own stylesheet.",
      },
    ],
    "info": Array [
      Object {
        "filePath": "./__tests__/multiple-files/other-component.scss",
        "line": null,
        "message": "Parsed",
      },
    ],
  },
}
`;
